### PR TITLE
Removed limit on input data size

### PR DESCRIPTION
**Pull request by YohDeadfall**
Perhaps the buffer was inherited from C++/WinRT, but haven't seen it for a long time where compile time generation of interface identifiers was heavily used. Anyway, it doesn't matter.

The change removes `ConstBuffer` and its limit put on an input. The price is a different `sha1` signature, it accepts a slice now.

### CODE REVIEW

**Review by rylev — DISMISSED** [review]
Thanks! I'm struggling to remember why I introduced the `ConstBuffer`. I thought it was because slice indexing was not allowed in older versions of Rust, but it seems like it's been supported for a while now.

**Comment by rylev** [comment]
Ah I found it! Slices cannot be resized in const contexts (i.e., `slice[0..10]` is not allowed). `ConstBuffer` is a hack to allow this. See: https://github.com/rylev/const-sha1/pull/4

**Comment by YohDeadfall** [comment]
So is it for external usage (win-rs)?

**Comment by rylev** [comment]
@YohDeadfall yes it's used [here](https://github.com/microsoft/windows-rs/blob/bf5efbf4933e3e44ff9322735065fa88d64d281c/crates/gen/src/type_name.rs#L133-L136). I wouldn't be opposed to adding an additional public function which just accepts a `&str`. It'd also be good to add documentation on why `ConstBuffer` is necessary.

**Comment by YohDeadfall** [comment]
Hello, @rylev! I returned back `ConstBuffer` with an annotation that it's a temporary replacement for slices in constant contexts and made unified code with `sha1` for slices.

File: src/lib.rs
Comment (rylev) [inline src/lib.rs]: Remove the term "sized".

File: src/lib.rs
Comment (rylev) [inline src/lib.rs]: Can you add docs saying when you should use this function vs the plain `sha1` function?

File: src/lib.rs
Comment (rylev) [inline src/lib.rs]: Not sure I like the name of the function. Perhaps `sha1_from_const_buffer` is better? Maybe we should rename `ConstBuffer` to `ConstSlice` and name this function `sha1_from_const_slice`. Thoughts?

**Review by rylev — CHANGES_REQUESTED** [review]
Thanks a lot! It's looking good, but I'd like a few changes before we merge.

File: src/lib.rs
Comment (YohDeadfall) [reply src/lib.rs]: That was a hardest part of the work, so I decided to leave it up to you, but forgot to mention it. Changing name to `sha1_from_const_slice` and renaming the struct sounds as the best solution.

**Review by rylev — APPROVED** [review]
Thank you!

**Pull request merged**
